### PR TITLE
Add RabbitMQ to order service

### LINKED ISSUE #49: Setup RabbitMQ messaging for Order Service

### PR DESCRIPTION
**Pull request by JortWillemsen**
## Changes
Implement RabbitMq for order service

## Checklist
- [x] Setup order service RabbitMq queue
- [x] Create Generic incoming event to test with
- [x] Test incoming events with RabbitMq dashboard
- [x] Restructure project  

## When merging, this pull request 
closes #49

### CODE REVIEW

File: delivery/src/main/java/nl/softwarestrijders/waiter/delivery/infrastructure/config/RabbitMqConfig.java [hunk 1]
```diff
@@ -22,25 +22,25 @@
     @Value("${messaging.exchange.waiter}")
     private String waiterExchangeName;
 
-    @Value("${messaging.queue.delivery}")
-    private String deliveryQueueName;
+    @Value("${messaging.queue.order}")
```
Comment (xandervedder) [inline delivery/src/main/java/nl/softwarestrijders/waiter/delivery/infrastructure/config/RabbitMqConfig.java:25]: This is not from your service fyi

File: order/src/main/java/nl/softwarestrijders/waiter/order/adapters/messaging/RabbitMqConfig.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+public class RabbitMqConfig {
+    @Value("${spring.rabbitmq.host}")
+    private String host;
+    @Value("${spring.rabbitmq.port}")
+    private int port;
+
+    @Value("${messaging.exchange.waiter}")
+    private String waiterExchangeName;
+    @Value("${messaging.queue.order}")
+    private String reviewMessageQueue;
```
Comment (xandervedder) [inline order/src/main/java/nl/softwarestrijders/waiter/order/adapters/messaging/RabbitMqConfig.java:23]: Should be ordermessagequeue

File: order/src/main/java/nl/softwarestrijders/waiter/order/adapters/messaging/RabbitMqConfig.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    @Value("${messaging.routing-key.order}")
+    private String reviewRoutingKey;
```
Comment (xandervedder) [inline order/src/main/java/nl/softwarestrijders/waiter/order/adapters/messaging/RabbitMqConfig.java:25]: Orderroutingkey

File: order/src/main/java/nl/softwarestrijders/waiter/order/adapters/messaging/RabbitMqEventListener.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+import org.springframework.amqp.rabbit.annotation.RabbitListener;
+import org.springframework.stereotype.Component;
+
+@Component
+public class RabbitMqEventListener {
+    @RabbitListener(queues = "#{'${messaging.queue.order}'}")
+    public void listen(GenericOrderEvent event) {
+        System.out.println("EVENT LOG: " + event.message());
+    }
+}
```
Comment (xandervedder) [inline order/src/main/java/nl/softwarestrijders/waiter/order/adapters/messaging/RabbitMqEventListener.java:13]: Nit: newline

**Review by xandervedder — CHANGES_REQUESTED** [review]

**Review by JustMilan — APPROVED** [review]

**Review by xandervedder — APPROVED** [review]

**Pull request merged**